x-pack/plugins/observability_solution/logs_explorer/public/components/data_source_selector/sub_components/data_view_filter.tsx

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState } from 'react';
+import {
+  EuiButtonEmpty,
+  EuiContextMenu,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPopover,
+  EuiText,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { DataViewFilterHandler } from '../types';
+import { DataViewsFilterParams } from '../../../state_machines/data_views';
+import {
+  allDataViewTypesLabel,
+  logsDataViewTypeLabel,
+  selectDataViewTypeLabel,
+} from '../constants';
+
+interface DataViewFilterProps {
+  onFilter: DataViewFilterHandler;
+  count: number;
+  filter: DataViewsFilterParams;
+}
+
+const logsDataViewType = 'logs';
+
+export const DataViewsFilter = ({ count, filter, onFilter }: DataViewFilterProps) => {
+  const [isPopoverOpen, setPopover] = useState(false);
+
+  const closeTypePopover = () => {
+    setPopover(false);
+  };
+
+  const togglePopover = () => {
+    setPopover(!isPopoverOpen);
+  };
+
+  const createSelectTypeFilter = (dataType: DataViewsFilterParams['dataType']) => {
+    return () => {
+      onFilter({ dataType });
+      closeTypePopover();
+    };
+  };
+
+  return (
+    <EuiFlexGroup alignItems="center" gutterSize="s" responsive={false}>
+      <EuiFlexItem grow={false}>
+        <EuiText color="subdued" size="xs">
+          {i18n.translate('xpack.logsExplorer.dataSourceSelector.dataViewCount', {
+            defaultMessage: '{count, plural, one {# data view} other {# data views}}',
+            values: { count },
+          })}
+        </EuiText>
+      </EuiFlexItem>
+      <EuiFlexItem grow={false}>
+        <EuiPopover
+          button={
+            <EuiButtonEmpty
+              data-test-subj="logsExplorerDataSourceSelectorDataViewTypeButton"
+              iconType="arrowDown"
+              iconSide="right"
+              size="xs"
+              onClick={togglePopover}
+            >
+              {filter.dataType !== logsDataViewType ? allDataViewTypesLabel : logsDataViewTypeLabel}
```

Comment:
**suggestion:** this doesn't scale automatically when we add/remove new filters, we could update it using a map such that:
```ts
function getSelectedFilterLabel(dataType: DataViewsFilterParams['dataType']) {
  const availableFilters = {
    [logsDataViewType]: logsDataViewTypeLabel
  }

  return availableFilters[dataType] || allDataViewTypesLabel
}

// ...

{getSelectedFilterLabel(filter.dataType)}
```